sidecar_acceptor.go

```diff
@@ -633,23 +609,51 @@ func (a *SidecarAcceptor) matchPrepare(pendingBatch *order.Batch,
 		},
 	})
 	if err != nil {
-		return nil, fmt.Errorf("error sending accept msg: %v", err)
+		return fmt.Errorf("error sending accept msg: %v", err)
 	}
 
-	return batch, nil
+	// We know we're involved in a batch, so let's store it for the
+	// next step.
+	a.pendingBatch = batch
+
+	return nil
+}
+
+// isPending returns true if the provided batchID matches the current pending
+// one.
+func (a *SidecarAcceptor) isPending(batchID []byte) bool {
+	if a.pendingBatch == nil || !bytes.Equal(batchID, a.pendingBatch.ID[:]) {
+		sdcrLog.Errorf("error processing batch sign message, unknown batch "+
+			"with ID %x", batchID)
+
+		return false
+	}
+
+	return true
 }
 
 // matchSign handles an incoming OrderMatchSignBegin message from the server.
 // Since we're only on the receiving end of a sidecar channel (which is always
 // a bid order) the tasks are simplified compared to normal bid order execution.
 //
 // NOTE: The lock must be held when calling this method.
-func (a *SidecarAcceptor) matchSign(batch *order.Batch) error {
+func (a *SidecarAcceptor) matchSign(
+	msg *auctioneerrpc.OrderMatchSignBegin) error {
+
+	// Assert we're in the correct state to receive a sign message.
+	if !a.isPending(msg.BatchId) {
+		return fmt.Errorf("pending batchID was: %x got: %x",
```

Comment:
Seems this would indicate a logic error elsewhere if we see it frequently in practice, or even a bug in the server's execution logic.

Reply:
Agreed, it should not be the normal flow. I have never seen it in our logs, but the code was already there and I think it does not hurt us to keep it. (I hope we end up in this state though*)